### PR TITLE
Fset 721Download the online test pass mark report including e-tray scores

### PR DESCRIPTION
**Pull request by mgilgar**
https://jira.tools.tax.service.gov.uk/browse/FSET-721

This story depends on storing the e-tray scores, which is not ready yet.

These are the PRs should be merged as part of this story:
https://github.com/hmrc/fset-faststream/pull/125
https://github.tools.tax.service.gov.uk/HMRC/fset-faststream-admin-frontend/pull/53

### CODE REVIEW

**Comment by ollyw** [comment]
@mgilgar is this story good to go from a functionality point of view? If it is, it would be good to get it merged so we can test today

**Comment by mgilgar** [comment]
@ollyw I have updated this story with the lastest from master in both faststream and faststream-admin-frontend. All tests including acceptance-tests run and pass locally, however it seems travis is struggling to run the test successfully. I am going to remove the labels so that you can merge if you wish.

File: app/repositories/application/GeneralApplicationRepository.scala [hunk 1]
```diff
@@ -493,6 +477,24 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
...
+      BSONDocument(s"progress-status.PHASE1_TESTS_RESULTS_RECEIVED" -> true)
+    ))
+
+    val projection = BSONDocument(
+      "userId" -> "1",
+      "applicationId" -> "1",
+      "scheme-preferences.schemes" -> "1",
+      "assistance-details" -> "1",
+      "testGroups" -> "1",
+      "progress-status" -> "2"
```
Comment (ollyw) [inline app/repositories/application/GeneralApplicationRepository.scala]: "2" doesn't seem to do anything. Suggest using true/false instead

File: app/model/report/CivilServiceExperienceDetailsReportItem.scala [hunk 2]
```diff
@@ -30,6 +30,7 @@ case class CivilServiceExperienceDetailsReportItem(
 object CivilServiceExperienceDetailsReportItem {
   implicit val civilServiceExperienceDetailsReportItemFormat = Json.format[CivilServiceExperienceDetailsReportItem]
 
+  // If you add a custom apply() to a case class companion object then Json.reads and Json.writes fail
```
Comment (mgilgar) [inline app/model/report/CivilServiceExperienceDetailsReportItem.scala]: That's good to know.

File: app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala [hunk 3]
```diff
@@ -107,8 +107,9 @@ trait GeneralApplicationRepoBSONToModelHelper {
     val applicationId = doc.getAs[String]("applicationId").getOrElse("")
     val progress: ProgressResponse = findProgress(doc, applicationId)
 
-    CandidateProgressReport(applicationId, Some(ProgressStatusesReportLabels.progressStatusNameInReports(progress)), schemes.getOrElse(List.empty[SchemeType]), disability, onlineAdjustments,
-      assessmentCentreAdjustments, gis, civilServant, fastTrack, edip, sdipPrevious, sdip, fastPassCertificate)
+    CandidateProgressReportItem(applicationId, Some(ProgressStatusesReportLabels.progressStatusNameInReports(progress)),
+      schemes.getOrElse(List.empty[SchemeType]), disability, onlineAdjustments, assessmentCentreAdjustments, gis, civilServant,
```
Comment (mgilgar) [inline app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala]: I will change it.

File: app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala [hunk 4]
```diff
@@ -178,32 +179,51 @@ trait GeneralApplicationRepoBSONToModelHelper {
     val onlineAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportForOnlineAssessment")).map(booleanTranslator)
     val assessmentCentreAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportAtVenue")).map(booleanTranslator)
 
+    val progress: ProgressResponse = findProgress(doc, applicationId)
+
+    ApplicationForOnlineTestPassMarkReport(
+      applicationId,
+      ProgressStatusesReportLabels.progressStatusNameInReports(progress),
+      schemes.getOrElse(List.empty[SchemeType]),
```
Comment (mgilgar) [inline app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala]: I will do it

File: app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala [hunk 5]
```diff
@@ -178,32 +179,51 @@ trait GeneralApplicationRepoBSONToModelHelper {
...
-      disability,
-      gis,
-      onlineAdjustments,
-      assessmentCentreAdjustments,
-      PassMarkReportTestResults(behaviouralTestResult, situationalTestResult))
+    val phase2DocOpt = testGroupsDoc.flatMap(_.getAs[BSONDocument]("PHASE2"))
+    val etrayTestResult = phase2DocOpt.flatMap { phase2Doc =>
+      val phase2TestProfile = Phase2TestGroup.bsonHandler.read(phase2Doc)
+      // TODO: review headOption
+      phase2TestProfile.activeTests.headOption.flatMap { phase2Test =>
```
Comment (mgilgar) [inline app/repositories/application/GeneralApplicationRepoBSONToModelHelper.scala]: I will do it

File: app/repositories/TestReportRepository.scala [hunk 6]
```diff
@@ -16,13 +16,13 @@
 
 package repositories
 
-import model.report.PassMarkReportTestResults
+import model.report.{TestResultsForOnlineTestPassMarkReportItem}
```
Comment (mgilgar) [inline app/repositories/TestReportRepository.scala]: I will fix it

File: app/repositories/application/GeneralApplicationRepository.scala (hunk 1 above)
Comment (mgilgar) [inline app/repositories/application/GeneralApplicationRepository.scala]: I change it to 1,we need it in the projection. I do not know what 2 means, it does not seem to be documented anywhere, we are using that in the other reports.

**Comment by mgilgar** [comment]
@ollyw I think all comments have been addressed. I have run the report and looks alright. It should be ready to merge.

**Comment by ollyw** [comment]
@mgilgar I just resolved some conflicts with master on this and will merge as soon as Travis is happy

**Comment by mgilgar** [comment]
@ollyw Thanks for resolving the conflicts, it must be something that was merged in the meantime.

**Pull request merged**